widget/segmentedentry.go

```diff
@@ -0,0 +1,230 @@
+package widget
+
+import (
+	"strings"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/theme"
+	"fyne.io/fyne/v2/widget"
+)
+
+var _ fyne.Focusable = (*SegmentedEntry)(nil)
+
+// SegmentedEntry shows a series of entries where each one only
+// accepts one value. You can optionally place a separator
+// between all or select fields.
+type SegmentedEntry struct {
+	widget.DisableableWidget
+
+	// A function to run when the grid is full
+	OnCompletion func(string)
+	// Text to use to separate fields. It will be
+	// forced to monospace to ensure consistent
+	// sizing.
+	Delimiter string
+	// Optionally specify which fields should have
+	// a separator after them.
+	DelimitAt []int
+
+	allowedRunes []rune
+	length       int
+
+	entries  []*SingleRuneEntry
+	selected int
+}
+
+// NewSegmentedEntry returns a new SegmentedEntry of the given length allowing
+// the given runes.
+func NewSegmentedEntry(allowedRunes []rune, length int) *SegmentedEntry {
+	// Initialize the input
+	input := &SegmentedEntry{
+		allowedRunes: allowedRunes,
+		length:       length,
+		entries:      make([]*SingleRuneEntry, length),
+		selected:     0,
+	}
+	input.ExtendBaseWidget(input)
+	return input
+}
+
+// GetValue returns the value of all the entries in the grid combined.
+func (g *SegmentedEntry) GetValue() string {
+	var out string
+	for _, e := range g.entries {
+		out += strings.TrimSpace(e.Text)
+	}
```

Comment:
The reason I didn't at first was the need for error checking on the [Write methods](https://pkg.go.dev/strings#Builder). I was worried it would piss off the linter in a function that under normal circumstances wouldn't need to worry about error checking.

Reply:
The write methods never returns any errors, check the  documentation. No need to handle them ;)

Reply:
But would the linter be as forgiving?